### PR TITLE
Source Monday: fix null record in response

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve https://github.com/airbytehq/oncall/issues/2471

## How
filter out null records from `response`. `LOG` message with empty records with possible error

## Recommended reading order
1. `extractor.py`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`8c937cca39`](https://github.com/airbytehq/airbyte/pull/29429/commits/8c937cca3905caab58812638068689da0a81b219)) - ✅

⏲️  Total pipeline duration: 04mn17s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-monday-oc-2471/1692104518/8c937cca3905caab58812638068689da0a81b219/source-monday/1.1.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:8c937cca3905caab58812638068689da0a81b219)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py [hunk 1]
```diff
@@ -15,6 +16,8 @@
 from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
 from airbyte_cdk.sources.declarative.types import Config, Record
 
+logger = logging.getLogger("airbyte")
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/extractor.py:19]: Should this be declared upper case? 
```
LOGGER = logging.getLogger()
```

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py [hunk 2]
```diff
@@ -89,6 +92,9 @@ def try_extract_records(self, response: requests.Response, field_path: List[Unio
             extracted = dpath.util.get(response_body, path, default=[])
 
         if extracted:
+            if isinstance(extracted, list) and None in extracted:
+                logger.warning(f"Record with null value received; errors: {response_body.get('errors')}")
+                return [x for x in extracted if x]
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/extractor.py:97]: Let's make a `generator` instead, if possible

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py (hunk 2 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-monday/source_monday/extractor.py:97]: Declarative CDK build upon simple functions, not the generators right now, need to rework CDK first.

**Pull request merged**